lib/app_profiler/middleware.rb

```diff
@@ -24,13 +25,31 @@ def call(env, params = AppProfiler::RequestParameters.new(Rack::Request.new(env)
 
     def profile(env, params)
       response = nil
+      orig_backend = nil
 
       return yield unless params.valid?
 
       params_hash = params.to_h
 
       return yield unless before_profile(env, params_hash)
 
+      @backend_lock.synchronize do
+        if params.backend &&
```

Comment:
I wonder why do we need this - if somehow we end up calling`profile` twice?

Reply:
My worry is Puma, if we have one thread that is being profiled by Stackprof, we cannot allow it to switch to Vernier while stackprof is running for instance.